File: src/DevServerClient/src/app/models/weather-forecast/weatherForecast.ts
import { Metadata } from "./metaData";
import { Units } from "./units";
import { ForecastDataPerHour } from "./forecastDataPerHour";
import { ForecastDataPerDay } from "./forecastDataPerDay";

export interface WeatherForecast {
    metadata: Metadata;
    units: Units;
    forecastDataPerDayPerHour: ForecastDataPerHour[];
    forecastDataPerDay: ForecastDataPerDay;
}

File: src/WeatherForecastApi/WeatherForecast/WeatherForecast.cs
using System.Text.Json.Serialization;

namespace WeatherForecastApi.WeatherForecast;

public record WeatherForecast(
    [property: JsonPropertyName("metadata")] Metadata Metadata,
    [property: JsonPropertyName("units")] Units Units,
    [property: JsonPropertyName("data_1h")] ForecastDataPerHour ForecastDataPerHour,
    [property: JsonPropertyName("data_day")] ForecastDataPerDay ForecastDataPerDay
    );